datafusion/core/tests/parquet/arrow_statistics.rs

```diff
@@ -507,9 +507,84 @@ async fn test_multiple_data_pages_nulls_and_negatives() {
 
 /////////////// MORE GENERAL TESTS //////////////////////
 // . Many columns in a file
-// . Differnet data types
+// . Different data types
 // . Different row group sizes
 
+// Float 16, 32, 64
+#[tokio::test]
+async fn test_float_16() {
+    let reader = TestReader {
+        scenario: Scenario::Float16,
+        row_per_group: 5,
+    }
+    .build()
+    .await;
+
+    Test {
+        reader: &reader,
+        expected_min: Arc::new(Float16Array::from(
+            vec![-5.0, -4.0, -0.0, 5.0]
+                .into_iter()
+                .map(f16::from_f32)
+                .collect::<Vec<_>>(),
+        )),
+        expected_max: Arc::new(Float16Array::from(
+            vec![-1.0, 0.0, 4.0, 9.0]
+                .into_iter()
+                .map(f16::from_f32)
+                .collect::<Vec<_>>(),
+        )),
+        expected_null_counts: UInt64Array::from(vec![0, 0, 0, 0]),
+        expected_row_counts: Some(UInt64Array::from(vec![5, 5, 5, 5])),
+        column_name: "f",
+        check: Check::RowGroup,
```

Comment:
I think this should be `Check::Both` ?

Reply:
Thanks, I think you're right, but doing that leads to a panic 🤔 ... gonna move the PR back draft while I look further into it.

Reply:
I think this is due to the difference in handling row_counts for RowGroups and DataPages. #10965

Im working on this - for now you'd need two test cases, one for each check with different expected row_counts.

Reply:
I'm stretching a here, but doing a bit of looking I think it's due to the physical type for an f16 being a fixed length byte array. I think `column_page_index_per_row_group_per_column` is an `Index::FIXED_LEN_BYTE_ARRAY` which seemingly isn't playing well with some of the conversion, so the values coming through end up being None.

Reply:
you seem to be right here; I left a comment in the code - where I think you might have missed something.

Reply:
FWIW https://github.com/apache/datafusion/issues/10965 is now fixex